back/engines/commercial/insights/app/jobs/insights/detect_categories_job.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module Insights
+  # Creates classification tasks to suggest categories for some inputs.
+  class DetectCategoriesJob < ::ApplicationJob
+    queue_as :default
+
+    # @param [Insights::View] view
+    def run(view)
+      category_detection_service ||= CategoryDetectionService.new
+      category_detection_service.detect_categories(view)
```

Comment:
```suggestion
      CategoryDetectionService.new.detect_categories(view)
```
(bc `category_detection_service` is a local var.)